src/chapter-03/src/core.rs

```diff
@@ -25,17 +13,20 @@ pub struct ShortenUrlResponse {
     shortened_url: String,
 }
 
-pub struct UrlShortener {}
+#[derive(Default, Debug)]
+pub struct UrlShortener {
+    urls: RwLock<HashMap<String, String>>,
```

Comment:
I removed the `Mutex` in favour of a `RwLock`. This allows us to have multiple readers at the same time. Only when writing the object is locked with unique access (which is generally a bit more performant in case of high throughput). The benefits of this approach in the context of Lambda are probably negligible, but I believe experienced Rust developers would be like "well, you should have used a `RwLock` here!"... so we might as well go with the general best practice.